homeassistant/components/foscam/__init__.py

```diff
@@ -81,3 +98,39 @@ def update_unique_id(entry):
     LOGGER.info("Migration to version %s successful", entry.version)
 
     return True
+
+
+class FoscamCoordinator(DataUpdateCoordinator):
+    """My custom coordinator."""
+
+    def __init__(self, hass, session, update_interval=timedelta(seconds=30)):
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            # Name of the data. For logging purposes.
+            name=DOMAIN,
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=update_interval,
+        )
+        self._session = session
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint."""
+        try:
+            async with async_timeout.timeout(30):
+                data = {}
+                data["dev_info"] = (
+                    await self.hass.async_add_executor_job(self._session.get_dev_info)
+                )[1]
```

Comment:
In the past, we've used `get_dev_info` for using the mac as the unique id and getting details like name and so and that broke the integration for many users as that info is only available if using a username and password that has admin rights (see #45921).

In fact, the return code you are ignoring here is `-3` when a non-admin account is being used, and the dictionary is empty, therefore you end with `data["dev_info"]` empty and when you try accessing it later it will throw a `KeyError`.

Forcing users to use admin accounts is not desirable and we rolled back those changes in order to be possible to configure the integration with a user with lower permissions (for example "operator").

```python
>>> fc = FoscamCamera("192.168.1.6", 88, "operator", "operator123", verbose=False)
>>> fc.get_dev_info()
(-3, OrderedDict())
```

Reply:
Ok, thanks for the info. I made the dev_info optional